File: test17_out/mcu.atmel.atmega/Timer0MatchA/Timer0MatchA.js

/*
 * ======== MODULE mcu.atmel.atmega.Timer0MatchA ========
 */

var $$c = function() {
    const Timer0MatchA = {};
    Timer0MatchA.Timer0MatchA = Timer0MatchA
    Timer0MatchA.$name = 'mcu.atmel.atmega.Timer0MatchA';
    Timer0MatchA.pollen$used = 0;
    Timer0MatchA.setHandler = new $$Ref('mcu_atmel_atmega_Timer0MatchA_setHandler__E');
    Timer0MatchA.enable = new $$Ref('mcu_atmel_atmega_Timer0MatchA_enable__E');
    Timer0MatchA.enabled = new $$Ref('mcu_atmel_atmega_Timer0MatchA_enabled__E');
    Timer0MatchA.disable = new $$Ref('mcu_atmel_atmega_Timer0MatchA_disable__E');
    Timer0MatchA.clear = new $$Ref('mcu_atmel_atmega_Timer0MatchA_clear__E');
    Timer0MatchA.isr = new $$Ref('mcu_atmel_atmega_Timer0MatchA_isr__I');
    Timer0MatchA.targetInit = new $$Ref('mcu_atmel_atmega_Timer0MatchA_targetInit__I');
    Timer0MatchA.name = "TIMER0_COMPA_vect";
    Timer0MatchA.enableRegister = "TIMSK0";
    Timer0MatchA.enableBit = "OCIE0A";
    Timer0MatchA.clearRegister = "TIFR0";
    Timer0MatchA.clearBit = "OCF0A";
    Timer0MatchA.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    Timer0MatchA.pollen__uses__ = function() {
    }
    Timer0MatchA.$$privateInit = function() {
        Timer0MatchA.h = 0;
    }
    return Timer0MatchA;
}
var $$u = $$c();
$units['mcu.atmel.atmega.Timer0MatchA'] = $$u;
$units.push($$u);

File: test17_out/mcu.atmel.atmega/Timer0MatchA/Timer0MatchA.c

/*
 * ======== string literals ========
 */

static const char mcu_atmel_atmega_Timer0MatchA__s__fileName[] = "Timer0MatchA.p";
static const char mcu_atmel_atmega_Timer0MatchA__s__1[] = "TIMER0_COMPA_vect";
static const char mcu_atmel_atmega_Timer0MatchA__s__3[] = "OCIE0A";
static const char mcu_atmel_atmega_Timer0MatchA__s__5[] = "OCF0A";
static const char mcu_atmel_atmega_Timer0MatchA__s__4[] = "TIFR0";
static const char mcu_atmel_atmega_Timer0MatchA__s__2[] = "TIMSK0";
static const char mcu_atmel_atmega_Timer0MatchA__s__0[] = "Timer0MatchA";

/*
 * ======== code injections ========
 */

 #include <avr/io.h> 
 #include <avr/interrupt.h> 
 ISR( 
TIMER0_COMPA_vect
 ) { 
isr()
;} 

/*
 * ======== function definitions ========
 */

void mcu_atmel_atmega_Timer0MatchA_setHandler__E( mcu_atmel_atmega_Timer0MatchA_Handler_handle h ) {
    mcu_atmel_atmega_Timer0MatchA_h__V = (h);
}

void mcu_atmel_atmega_Timer0MatchA_enable__E() {
    TIMSK0 |= (1 << OCIE0A);
}

void mcu_atmel_atmega_Timer0MatchA_disable__E() {
    TIMSK0 &= ~(1 << OCIE0A);
}

void mcu_atmel_atmega_Timer0MatchA_clear__E() {
    TIFR0 &= ~(1 << OCF0A);
}

void mcu_atmel_atmega_Timer0MatchA_isr__I() {
    if ((mcu_atmel_atmega_Timer0MatchA_h__V) != null) {
        mcu_atmel_atmega_Timer0MatchA_h__V();
    }
}

void mcu_atmel_atmega_Timer0MatchA_targetInit__I() {
}

bool mcu_atmel_atmega_Timer0MatchA_enabled__E() {
    return(  TIMSK0 & (1 << OCIE0A)  );
}

